### ISSUE TITLE
Stuck in blank screen after launch

### ISSUE THREAD

**Issue by gmq** [salomvary/soundcleod#105]
The first time I launched the app it worked okay. Every time after that it gets stuck in a white screen with the following message in the system log:
> 6/4/16 12:41:08.090 SoundCleod[66133]: *** -[NSPlaceholderString initWithFormat:locale:arguments:]: nil argument

Deleting SoundCleod's .plist file makes it work again.

**Comment by felipeptx** [comment]
It's happening exactly the same with me. Running on El Capitan 10.11.5

**Comment by felipeptx** [comment]
Where can I find this .plist file?

**Comment by gmq** [comment]
@felipeptx `~/Library/Preferences/sc.SoundCleod.plist`.

**Comment by felipeptx** [comment]
Thank you very much! 
I know it might not be the most appropriate place to say that, but the very best option I have found for a soundcloud client for mac is Cumulus which can be downloaded right here: https://gillesdemey.github.io/Cumulus/
I spent a plenty of time seeking a good client to listen to my soundcloud playlists, but always a good tool was missing. And Cumulus seems to be the most complete app available for free. You can easily access your playlists and so on. hope being helpful
cheers

**Comment by salomvary** [comment]
More information on this might help me investigating:

- Does this happen on anything else than El Capitan?
- Is is related to OSX language settings (or everyone using US English?)

As I don't have El Capitan installed at the moment, it might take a while to investigate. Volunteer help welcome :)

**Comment by salomvary** [comment]
> I know it might not be the most appropriate place to say that, but the very best option I have found for a soundcloud client for mac is Cumulus

@felipeptx No worries, it's already listed as an alternative ;)
https://github.com/salomvary/soundcleod#alternatives

**Comment by accatyyc** [comment]
I'm unable to reproduce this on debug builds. Maybe it only happens after updating from the previous version?

**Comment by salomvary** [comment]
> Maybe it only happens after updating from the previous version?

That's an interesting hypothesis. Could someone who is still having issues share their SoundCleod.app? (eg. `zip -r ~/Desktop/SoundCleod.zip /Applications/SoundCleod.app` and share it  somehow?)

**Comment by gmq** [comment]
> Maybe it only happens after updating from the previous version?

First time I install the app, haven't updated it yet.

> Is is related to OSX language settings (or everyone using US English?)

![screen shot 2016-06-06 at 10 12 07](https://cloud.githubusercontent.com/assets/472791/15824838/2685b14c-2bcf-11e6-9375-d0a9b7c1acdc.png)

> Could someone who is still having issues share their SoundCleod.app? (eg. zip -r ~/Desktop/SoundCleod.zip /Applications/SoundCleod.app and share it somehow?)

[SoundCleod.zip](https://github.com/salomvary/soundcleod/files/300573/SoundCleod.zip)

**Comment by salomvary** [comment]
Weird. Haven't managed to reproduce the issue myself:
- Running on El Capitan
- Tried both English and non-English locale
- Tried old install and fresh install

**Comment by MrUnknownGuyAC** [comment]
Having the same issue here... Cleared .plist file, didn't work. Uninstalled with AppCleaner, didn't work.

All this was from the downloaded version from the "Download from here" link on the main page.

I have tried the hombrew cask download version, and it seems that it has found the file has two different MD5 hases.

```
$ brew cask install soundcleod

==> Downloading https://github.com/salomvary/soundcleod/raw/master/dist/SoundCle
Already downloaded: /Users/__________/Library/Caches/Homebrew/soundcleod-0.18.dmg
==> Verifying checksum for Cask soundcleod
==> Note: running "brew update" may fix sha256 checksum errors
Error: sha256 mismatch
Expected: 467761e18935787f2b1b89179236a31162521975b5104dfecb0869aec15403c9
Actual: 5f7bad26b2bb7b875787d63d4a0fd1cbfc4c9edf998a1e265fcc775a4b4b4eaf
File: /Users/__________/Library/Caches/Homebrew/soundcleod-0.18.dmg
To retry an incomplete download, remove the file above.
```

As for line 4 (already downloaded), it was not downloaded before hand.
After the result, I tried again after deleting the file with the same results.

**Comment by salomvary** [comment]
Ok, managed to reproduce the problem, very bizarre:

- Used a fresh install of El Capitan
- Installed SoundCleod from the latest released dmg
- SoundCleod started with a blank screen
- Installed XCode and enabled developer mode to debug, the problem is gone. Both when executed from /Applications or XCode
- Disabled developer mode, but SoundCleod still starts fine

I have no idea what this might be :(

**Comment by salomvary** [comment]
I think the problem was an outdated version of Sparkle framework (managing automatic updates).

Just released SoundCleod 0.20 which fortunately fixed this issue. Unfortunately the automatic updates are borked on El Capitan (and maybe others?), **SoundCleod has to be reinstalled from a fresh download**: http://soundcleod.com/
